### PR TITLE
[PREVIEW] Feature/sl 1551/search and find listing request

### PR DESCRIPTION
**Pull request by kpakur**
Main Parts are HearingQuery.java that contains the creation of the complex sql query with subqueries in select and also in where clause.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingService.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                && operation.equals(ComparisonOperations.EQUALS)) {
+                boolean isListed = criteria.getValue().toString().equals("listed");
+
+                Subquery<HearingPart> subquery = cq.subquery(HearingPart.class);
+                Root<HearingPart> hpRoot = subquery.from(HearingPart.class);
+                subquery.select(hpRoot);
+                if (isListed) {
+                    subquery.where(cb.equal(hpRoot.get(HearingPart_.hearingId), hearingRoot),
+                        cb.isNotNull(hpRoot.get(HearingPart_.sessionId)));
+                } else {
```
Comment (zielinskikamil) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingService.java]: oh shit ;/

Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingService.java]: @zielinskikamil wtf?

Comment (zielinskikamil) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingService.java]: I've just imagined that I need to change sth here ^^. Looks very busy but tbh I don't have a clue how we can split it.

File: build.gradle [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
   id 'application'
+  id "java"
   id 'checkstyle'
   id 'pmd'
   id 'jacoco'
   id 'io.spring.dependency-management' version '1.0.4.RELEASE'
   id 'org.springframework.boot' version '1.5.10.RELEASE'
   id 'org.owasp.dependencycheck' version '3.1.1'
   id 'com.github.ben-manes.versions' version '0.17.0'
+//  id "at.comm_unity.gradle.plugins.jpamodelgen" version "1.1.4"
```
Comment (zielinskikamil) [inline build.gradle]: can we remove this comment?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/repository/queries/HearingQueries.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+@Component
+public class HearingQueries {
+
+    @Autowired
+    HearingRepository hearingRepository;
+
+    @Autowired
+    private EntityManager entityManager;
+
```
Comment (zielinskikamil) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/repository/queries/HearingQueries.java]: plz remove unnecessary lines

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/repository/queries/HearingQueries.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    }
+
+    private Predicate createInPredicate(Root<Hearing> hearingRoot, SearchCriteria criteria) {
+        String key = criteria.getKey();
+        Path<Object> rootKey = null;
+        if (key.contains(".")) {
+            rootKey = hearingRoot.get(key.split("\\.") [0]).get(key.split("\\.") [1]);
+        } else {
+            rootKey = hearingRoot.get(key);
+        }
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/repository/queries/HearingQueries.java]: Lines above and code in createInOrNullPredicate method could be extracted probably?

File: src/main/resources/application.yaml [hunk 5]
```diff
@@ -85,7 +85,7 @@ spring:
     properties:
       hibernate:
         dialect: org.hibernate.dialect.PostgreSQLDialect
-    show-sql: false
+    show-sql: true
```
Comment (r-kruk-kainos) [inline src/main/resources/application.yaml]: Is it here on purpose?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/repository/queries/HearingQueries.java (hunk 3 above)
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/repository/queries/HearingQueries.java]: done

File: src/main/resources/application.yaml (hunk 5 above)
Comment (kpakur) [reply src/main/resources/application.yaml]: changing to false

File: build.gradle [hunk 6]
```diff
@@ -186,12 +179,12 @@ task smoke(type: Test, description: 'Runs the smoke tests.', group: 'Verificatio
...
 }
 
-task functional(type: Test, description: 'Runs the functional tests.', group: 'Verification') {
+task functional(type: Test, description: 'Runs the functional tests on a running application.', group: 'Verification') {
   testClassesDirs = sourceSets.functional.output.classesDirs
   classpath = sourceSets.functional.runtimeClasspath
 }
 
-task integrationFakeRulesTests(type: Test, description: 'Runs the integration tests.', group: 'Verification') {
+task integrationFakeRulesTests(type: Test, description: 'Runs the integration tests iwth wiremock.', group: 'Verification') {
```
Comment (leszekgu) [inline build.gradle]: typo (iwth)

Comment (kpakur) [reply build.gradle]: done

**Pull request merged**